### PR TITLE
Use helper and webdav host from settings

### PR DESCRIPTION
**Pull request by loicgasser**
Fix for https://github.com/geoadmin/mf-chsdi3/issues/1613

This PR adds a new variable in settings:
webdav_host

We also use a helper function to determine if a resource exists or not. (We request only the header)

In general I find it really annoying that we need to check for the existence of a resource on webdav. If a link exists in DB we should have it on webDav or we don't put the link in the DB to start with.

I am not changing that everywhere for now as we have ton of references.

@AFoletti @daguer @ltclm @pfanguin @alcapat 

Anyone motivated to change that in all templates?

Simply do:
git grep dav0.bgdi.admin.ch

and follow what has been done in this PR. (using helper function and variable from settings)

### CODE REVIEW

**Comment by pfanguin** [comment]
I can do it !

File: chsdi/templates/htmlpopup/ivs_nat.mako [hunk 1]
```diff
@@ -32,52 +32,57 @@
...
+        PDF_Level_2_exist = PDF_Full[6:8]
+        PDF_Level_2 = PDF_Full[0:8] + '00'
+        PDF_Level_2_Name = PDF_Level_1_Name + '.' +  str(int(PDF_Full[6:8]))
+        PDF_Level_3_exist = PDF_Full[8:10] 
+        PDF_Level_3 = PDF_Full 
+        PDF_Level_3_Name = PDF_Level_2_Name + '.' + str(int(PDF_Full[8:10]))
+        url = webDavHost + "/kogis_web/downloads/ivs/beschr/de/" + c['attributes']['ivs_sortsla'] + ".pdf"
+        pdf = resource_exists(url)
+        response = None
+        try:
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/ivs_nat.mako]: you don't need this part anymore as you're using:
`pdf = resource_exists(url)` just before

File: chsdi/templates/htmlpopup/geotechnik_mineralische_rohstoffe200.mako [hunk 2]
```diff
@@ -2,13 +2,19 @@
 
 <%def name="table_body(c,lang)">
 <%
-    PDF_Link = 'https://dav0.bgdi.admin.ch/kogis_web/downloads/geologie/geotechnik/' + c['attributes']['legend'] 
+
+    from chsdi.lib.helpers import resource_exists
+    pdf = None
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/geotechnik_mineralische_rohstoffe200.mako]: I don't think you need to test for the existence of the resource (as it was not the case previously) or if you do, you should use the varaible `pdf` you define in your mako.

File: chsdi/templates/htmlpopup/geotope.mako [hunk 3]
```diff
@@ -3,5 +3,13 @@
 <%def name="table_body(c,lang)">
 <tr><td class="cell-left">${_('name')}</td><td>${c['attributes']['nom'] or '-'}</td></tr>
 <tr><td class="cell-left">${_('nummer')}</td><td>${c['attributes']['nummer'] or '-'}</td></tr>
-<tr><td class="cell-left">${_('link2dok')}</td>    <td><a href="https://dav0.bgdi.admin.ch/kogis_web/downloads/geologie/geotope/geotope-CH_${c['attributes']['fix_id']}.pdf" target="_blank">Link</a></td></tr>
+<%
+    from chsdi.lib.helpers import resource_exists
+    pdf = None
+    if c['attributes']['fix_id'] is not None:
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/geotope.mako]: same remark concerning pdf variable

File: chsdi/templates/htmlpopup/geotechnik_gk200.mako [hunk 4]
```diff
@@ -2,13 +2,18 @@
 
 <%def name="table_body(c,lang)">
 <%
-    PDF_Link = 'https://dav0.bgdi.admin.ch/kogis_web/downloads/geologie/geotechnik/' + c['attributes']['file_name'] + '_' + lang + '.pdf'
+    from chsdi.lib.helpers import resource_exists
+    pdf = None
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/geotechnik_gk200.mako]: same remark concerning pdf variable

File: chsdi/templates/htmlpopup/geschiebemessnetz.mako [hunk 5]
```diff
@@ -11,10 +11,18 @@
...
+    from chsdi.lib.helpers import resource_exists
+    pdf = None
+    if c['attributes']['pdf_file'] is not None:
+        webDavHost = request.registry.settings['webdav_host']
+        url_pdf = webDavHost + '/kogis_web/downloads/bafu/geschiebemessnetz/' + c['attributes']['pdf_file']
+        pdf = resource_exists(url_pdf)
+%>
 <tr><td class="cell-left">${_('link')}</td>
     <td>
     % if c['attributes']['pdf_file']:
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/geschiebemessnetz.mako]: use `if pdf:`

as this makes sure the pdf exists.

File: chsdi/templates/htmlpopup/kantone.ivs-reg_loc.mako [hunk 6]
```diff
@@ -25,54 +25,59 @@
...
+        PDF_Level_2_exist = PDF_Full[6:8]
+        PDF_Level_2 = PDF_Full[0:8] + '00'
+        PDF_Level_2_Name = PDF_Level_1_Name + '.' +  str(int(PDF_Full[6:8]))
+        PDF_Level_3_exist = PDF_Full[8:10] 
+        PDF_Level_3 = PDF_Full 
+        PDF_Level_3_Name = PDF_Level_2_Name + '.' + str(int(PDF_Full[8:10]))
+        url = webDavHost +  "/kogis_web/downloads/ivs/beschr/de/"+c['attributes']['ivs_sortsla']+".pdf"
+        pdf = resource_exists(url)
+        response = None
+        try:
```
Comment (loicgasser) [inline chsdi/templates/htmlpopup/kantone.ivs-reg_loc.mako]: you can remove this part

**Comment by loicgasser** [comment]
I think you also missied `chsdi/templates/htmlpopup/energiestaedte_2000watt_areale.mako`
Thanks a lot for this though.

**Comment by loicgasser** [comment]
ok I did it finally.

**Pull request merged**